kliff/dataset/dataset.py

```diff
@@ -184,16 +195,45 @@ def stress(self) -> List[float]:
     @property
     def weight(self):
         """
-        Get the weight of the configuration if the loss function.
+        Get the weight class of the loss function.
         """
         return self._weight
 
+    @property
```

Comment:
We'd expect all kinds of weight be accessed via self.weight, instead of the config_weight, ... stress_weight properties. So, I guess the four properties can be deleted?

Reply:
Does this mean if user wants to access the weight, he/she would do `config.weight.energy_weight`, etc.?

Reply:
Yes. I believe this is a bit cleaner, instead of attaching all the stuff to the Configuration class.